src/core/symbology/classification/qgsclassificationmethod.cpp

```diff
@@ -0,0 +1,300 @@
+/***************************************************************************
+    qgsclassificationmethod.cpp
+    ---------------------
+    begin                : September 2019
+    copyright            : (C) 2019 by Denis Rouzaud
+    email                : denis@opengis.ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgis.h"
+#include "qgsclassificationmethod.h"
+#include "qgsvectorlayerutils.h"
+#include "qgsvectorlayer.h"
+#include "qgsgraduatedsymbolrenderer.h"
+#include "qgsapplication.h"
+#include "qgsclassificationmethodregistry.h"
+
+const int QgsClassificationMethod::MAX_PRECISION = 15;
+const int QgsClassificationMethod::MIN_PRECISION = -6;
+
+static const QRegExp RE_TRAILING_ZEROES = QRegExp( "[.,]?0*$" );
```

Comment:
QRegExp -> QRegularExpression

Reply:
Also am wondering -- is this thread safe? There's a strong requirement to ensure that the classification methods ARE, so that they can be used in processing.